### ISSUE TITLE
Add formatter-compatible rule to enforce one-arg-per-line + trailing comma when line-length exceeded

### ISSUE THREAD

**Issue by zerlok** [astral-sh/ruff#19308]
### Summary

**Current limitation**:

* Ruff formatter reflows function signatures when line-length is exceeded
* However, it preserves grouped arguments even when broken across lines
* It does **not** insert trailing commas
* `COM812` adds trailing commas only if the layout is already multiline
* One-arg-per-line layout only occurs **after** trailing comma is inserted

This results in a 3-pass cycle to achieve desired layout:

1. Format → break long line
2. Fix → add comma
3. Format → apply one-arg-per-line

**Request**:

Add a rule or formatter enhancement to enforce:

* If function signature exceeds `line-length`

  * Break into multiline
  * Place one argument per line
  * Add trailing comma
* If it does **not** exceed line-length, keep as-is

**Goal**:

Allow this transformation to occur in **one pass**, avoiding chained invocations and repeated file watcher triggers.

**Benefit**:

Cleaner formatting, aligned with Black-like behavior but under Ruff’s control, and better editor automation compatibility.

**Example**:

```python
# Input
def f(a: int, b: int, c: int, d: int, e: int) -> int:
    return a + b + c + d + e

# Desired output
def f(
    a: int,
    b: int,
    c: int,
    d: int,
    e: int,
) -> int:
    return a + b + c + d + e
```

**Related**:

* [https://github.com/astral-sh/ruff/issues/6357](https://github.com/astral-sh/ruff/issues/6357)
* [https://github.com/astral-sh/ruff/issues/10966](https://github.com/astral-sh/ruff/issues/10966)
